cli_tools/gce_ovf_export/exporter/flags.go

```diff
@@ -0,0 +1,78 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package ovfexporter
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/daisy"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/flags"
+	ovfexportdomain "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/gce_ovf_export/domain"
+)
+
+// RegisterFlags registers OVF exporter CLI flags with args
+func RegisterFlags(ovfExportArgs *ovfexportdomain.OVFExportArgs, args []string) error {
+	flagSet := flag.NewFlagSet("ovf-export", flag.ContinueOnError)
+	// Don't write parse errors to stdout, instead propagate them via an
+	// exception since we use flag.ContinueOnError.
+	flagSet.SetOutput(ioutil.Discard)
+
+	flagSet.Var((*flags.TrimmedString)(&ovfExportArgs.InstanceName), ovfexportdomain.InstanceNameFlagKey,
+		"Name of the VM Instance to be exported.")
+	flagSet.Var((*flags.TrimmedString)(&ovfExportArgs.MachineImageName), ovfexportdomain.MachineImageNameFlagKey,
+		"Name of the Google machine image to be exported.")
+	flagSet.Var((*flags.LowerTrimmedString)(&ovfExportArgs.ClientID), ovfexportdomain.ClientIDFlagKey,
+		"Identifies the client of the exporter, e.g. `gcloud` or `pantheon`")
+	flagSet.Var((*flags.TrimmedString)(&ovfExportArgs.ClientVersion), "client-version",
+		"Identifies the version of the client of the exporter.")
+	flagSet.Var((*flags.TrimmedString)(&ovfExportArgs.DestinationURI), ovfexportdomain.DestinationURIFlagKey,
+		"Google Cloud Storage URI of the OVF or OVA file to export to. For example: gs://my-bucket/my-vm.ovf.")
```

Comment:
Extra trailing dot?

In terms of UX, if the user specifies ova, do they need to provide a path that is file-like and ends with ".ova"? In other words, is this an error?

`-destination-uri gs://bucket/ -ovf-format ova`

Reply:
Re extra trailing dot, it's intended to be end of a sentence. 

OVF descriptor has only one format, so OVF format here means the format of the whole OVF package, whether it's unpacked (OVF), or packed (OVA). 

Looking at VMWare's ovftool, I think I'll make the ovf/ova selection automatic, based on destination file path (OVA if ending with ".ova", OVF otherwise.

Reply:
Removed -ovf-format and improved logic around destination-uri and exported file names